tests/py/unit/test_build_steps.py

```
@@ -0,0 +1,1078 @@
+        time.sleep(0.01)
+        output_js = dist_dir / "bundle.js"
+        output_js.write_text("// built")
+
+        step_ran = []
+
+        class TrackingStep(BuildStep):
+            @property
+            def name(self) -> str:
+                return "tracking"
+
+            def run(self, ctx: BuildContext) -> None:
```

Ruff (0.14.13)

Unused method argument: `ctx`

(ARG002)